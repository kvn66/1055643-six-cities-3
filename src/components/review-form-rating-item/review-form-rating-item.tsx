import * as React from "react";

type Props = {
  id: number;
  title: string;
  rating: number;
  formIsLocked: boolean;
  onChangeRating: (evt: { target: { value: string } }) => void;
}

const ReviewFormRatingItem: React.FunctionComponent<Props> = (props: Props) => {
  const {id, title, rating, formIsLocked, onChangeRating} = props;

  return (
    <React.Fragment>
      <input
        onChange={onChangeRating}
        className="form__rating-input visually-hidden" name="rating" value={`${id}`} id={`${id}-stars`} type="radio"
        disabled={formIsLocked}
        checked={rating === id}
      />
      <label htmlFor={`${id}-stars`} className="reviews__rating-label form__rating-label" title={title}>
        <svg className="form__star-image" width="37" height="33">
          <use xlinkHref="#icon-star"/>
        </svg>
      </label>
    </React.Fragment>
  );
};

export const MemoizedReviewFormRatingItem = React.memo(ReviewFormRatingItem);

export default ReviewFormRatingItem;
