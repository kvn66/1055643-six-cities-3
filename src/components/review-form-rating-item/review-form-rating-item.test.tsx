import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {MemoizedReviewFormRatingItem} from "./review-form-rating-item";
import {ReactTestRendererJSON} from "react-test-renderer";

const testFn: jest.Mock = jest.fn();

it(`Render MemoizedReviewFormRatingItem`, () => {
  const tree: ReactTestRendererJSON = renderer
    .create(
        <MemoizedReviewFormRatingItem
          id={1}
          title={`terribly`}
          rating={4}
          formIsLocked={false}
          onChangeRating={testFn}
        />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
