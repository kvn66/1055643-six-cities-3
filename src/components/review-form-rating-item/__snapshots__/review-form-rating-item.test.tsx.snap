// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render MemoizedReviewFormRatingItem 1`] = `
Array [
  <input
    checked={false}
    className="form__rating-input visually-hidden"
    disabled={false}
    id="1-stars"
    name="rating"
    onChange={[MockFunction]}
    type="radio"
    value="1"
  />,
  <label
    className="reviews__rating-label form__rating-label"
    htmlFor="1-stars"
    title="terribly"
  >
    <svg
      className="form__star-image"
      height="33"
      width="37"
    >
      <use
        xlinkHref="#icon-star"
      />
    </svg>
  </label>,
]
`;
